File: ois/2021/recipe/main.cpp
#include <bits/stdc++.h>

using namespace std;

int n_ingredients, m_recipes;
double total_weights[1005];
int weights[1005];

int main () {
    cin >> n_ingredients >> m_recipes;
    for (int i = 0; i < m_recipes; i++) {
        double normalizer = 0;
        for (int j = 0; j < n_ingredients; j++) {
            cin >> weights[j];
            normalizer += weights[j];
        }
        normalizer = 1000000 / normalizer;

        for (int j = 0; j < n_ingredients; j++) {
            total_weights[j] += weights[j] * normalizer;
        }
    }

    for (int j = 0; j < n_ingredients; j++) {
        cout << (int)(total_weights[j] / m_recipes) << " ";
    }
    cout << "\n";
}

File: ois/2021/recipe/main.py
n_ingredients, m_websites = (int(x) for x in input().split())

total_weights = [0 for _ in range(n_ingredients)]
NORMALIZED_WEIGHT = 10 ** 6

for _ in range(m_websites):
    weights = [int(x) for x in input().split()]
    weights_sum = sum(weights)
    # normalize to 1Kg
    normalizer = NORMALIZED_WEIGHT / weights_sum

    for i in range(n_ingredients):
        total_weights[i] += weights[i] * normalizer

avg_weights = (int(weight / m_websites) for weight in total_weights)
print(' '.join(str(x) for x in avg_weights))
